src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/jvm/sizing/HeapSizeIncreasePolicy.java

```diff
@@ -0,0 +1,142 @@
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+
+public class HeapSizeIncreasePolicy implements DecisionPolicy {
+
+  private final LargeHeapClusterRca largeHeapClusterRca;
+  private AppContext appContext;
+  private RcaConf rcaConf;
+  private final HeapSizeIncreaseClusterMonitor heapSizeIncreaseClusterMonitor;
+
+  private int unhealthyNodePercentage;
+
+  public HeapSizeIncreasePolicy(final LargeHeapClusterRca largeHeapClusterRca) {
+    this(largeHeapClusterRca, null);
+  }
+
+  public HeapSizeIncreasePolicy(final LargeHeapClusterRca largeHeapClusterRca,
+      final HeapSizeIncreaseClusterMonitor clusterMonitor) {
+    this.largeHeapClusterRca = largeHeapClusterRca;
+    if (clusterMonitor != null) {
+      this.heapSizeIncreaseClusterMonitor = clusterMonitor;
+    } else {
+      this.heapSizeIncreaseClusterMonitor = new HeapSizeIncreaseClusterMonitor();
+    }
+  }
+
+  @Override
+  public List<Action> evaluate() {
+    addToClusterMonitor();
+
+    List<Action> actions = new ArrayList<>();
+    if (!heapSizeIncreaseClusterMonitor.isHealthy()) {
+      Action heapSizeIncreaseAction = new HeapSizeIncreaseAction(appContext);
+      if (heapSizeIncreaseAction.isActionable()) {
+        actions.add(heapSizeIncreaseAction);
+        return actions;
+      }
+    }
+
+    return actions;
+  }
+
+  private void addToClusterMonitor() {
+    long currTime = System.currentTimeMillis();
+    if (largeHeapClusterRca.getFlowUnits().isEmpty()) {
+      return;
+    }
+    ResourceFlowUnit<HotClusterSummary> flowUnit = largeHeapClusterRca.getFlowUnits().get(0);
+
+    if (flowUnit.getSummary() == null) {
+      return;
+    }
+    List<HotNodeSummary> hotNodeSummaries = flowUnit.getSummary().getHotNodeSummaryList();
+    hotNodeSummaries.forEach(hotNodeSummary -> {
+      NodeKey nodeKey = new NodeKey(hotNodeSummary.getNodeID(), hotNodeSummary.getHostAddress());
+      heapSizeIncreaseClusterMonitor.recordIssue(nodeKey, currTime);
+    });
+  }
+
+  private class HeapSizeIncreaseClusterMonitor {
+
+    private static final int DEFAULT_DAY_BREACH_THRESHOLD = 8;
+    private static final int DEFAULT_WEEK_BREACH_THRESHOLD = 3;
+    private final Map<NodeKey, AlarmMonitor> perNodeMonitor;
+
+    HeapSizeIncreaseClusterMonitor() {
+      this.perNodeMonitor = new HashMap<>();
+    }
+
+    public void recordIssue(final NodeKey nodeKey, long currTimeStamp) {
+      perNodeMonitor.computeIfAbsent(nodeKey,
+          key -> new JvmActionsAlarmMonitor(DEFAULT_DAY_BREACH_THRESHOLD,
+              DEFAULT_WEEK_BREACH_THRESHOLD))
+                    .recordIssue(currTimeStamp, 1d);
+    }
+
+    public boolean isHealthy() {
+      int numDataNodesInCluster = appContext.getDataNodeInstances().size();
+      double unhealthyCount = 0;
+      for (Map.Entry<NodeKey, AlarmMonitor> entry : perNodeMonitor.entrySet()) {
+        if (!entry.getValue().isHealthy()) {
+          unhealthyCount++;
+        }
+      }
```

Comment:
nit: since we are not using the key here, can we simplify it with a for each loop on just the values?

Reply:
done.